GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,173 @@
+    this._fontTobeUnloaded.splice(fontCachePosition, 1);
+  }
+};
+
+/**
+ * When an bitmapText object is removed, decrease the count of objects related to the font in the manager.
+ * When a font is not unused anymore, it goes in a temporary cache. The cache holds up to 10 fonts.
+ * If the cache reaches its maximum capacity, the oldest font is unloaded from memory.
+ * @param {string} name Slug of the bitmapFont
+ */
+gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
+  if (!this._fontUsed[name]) {
+    // We tried to remove font that was never marked as used.
+    console.error(
+      'BitmapFont with name ' +
+        name +
+        ' was tried to be removed but was never marked as used.'
+    );
+    return;
+  }
+
+  this._fontUsed[name].objectsUsingTheFont--;
+
+  if (this._fontUsed[name].objectsUsingTheFont <= 0) {
+    delete this._fontUsed[name];
+
+    // Add the font name at the last position of the cache.
+    if (!this._fontTobeUnloaded.includes(name))
+      this._fontTobeUnloaded.push(name);
+
+    if (this._fontTobeUnloaded.length > 10) {
+      // Remove the first font (i.e: the oldest one)
+      const oldestUnloadedFont = this._fontTobeUnloaded.shift();
+      PIXI.BitmapFont.uninstall(oldestUnloadedFont);
+      console.log('Uninstall bitmapFont: ' + oldestUnloadedFont);
+    }
+  }
+};
+
+// This method is asynchronous and return a promise with the fontName.
+gdjs.BitmapFontManager.prototype.loadBitmapFont = async function (
+  bitmapFontResourceName,
+  texture
+) {
+  let bitmapFontResource = this._resources.find(function (resource) {
+    return (
+      resource.kind === 'bitmapFont' && resource.name === bitmapFontResourceName
+    );
+  });
+
+  if (!bitmapFontResource) {
+    return Promise.reject(
+      new Error(
+        'Can\'t find resource with name: "' +
+          bitmapFontResourceName +
+          '" (or is not a bitmapFont file).'
+      )
+    );
+  }
+
+  let response = await fetch(bitmapFontResource.file);
+  if (!response.ok) {
+    return Promise.reject(
+      new Error(
+        "Can't fetch the bitmap font file, error: " +
+          response.status +
+          '(' +
+          response.statusText +
+          ')'
+      )
+    );
+  }
+
+  let fontData = await response.text();
+  let bitmapFont = PIXI.BitmapFont.install(fontData, texture);
+  return Promise.resolve(bitmapFont.font);
+};
+
+/**
+ * Check if a bitmap font exist by this name.
+ * The font name is defined in the font file (.fnt/.xml)
+ * @param {string} fontName Name of the bitmap font
+ */
+gdjs.BitmapFontManager.prototype.isBitmapFontLoaded = function (fontName) {
+  return !!PIXI.BitmapFont.available[fontName];
+};
+
+// function not yet used, because not complete
+// At the end loadBitmapFont() must use as arguments bitmapFontResourceName and texture
+// But texture isn't yet setup in this function
+gdjs.BitmapFontManager.prototype.preloadBitmapFonts = function (
```

Comment:
This "resource manager" is a bit special, that's why you struggled with this function :)

Other resource manager are "simple". They load a single thing: a texture file, a font file, a json file, an audio file...

This resource manager is a bit different. Basically, it offers you a function that takes:
- a bitmapFont resource name
- a texture name

and it install it in memory (by calling `BitmapFont.install`) and returns it to you. But for this, we need to have two things loaded:
- the texture obviously, but it's easy because it's loaded by the `PixiImageManager` already.
- and the bitmap font data! :)

So in this function, we don't really preload bitmap fonts here in the Pixi.js sense. We preload files, and these files are the bitmap font data (what you called `fontData`).

So I suggest:
- Rename this function preloadBitmapFontData = function(...) { ... }
- Don't even try to call `loadBitmapFont`. Because we don't have textures. Because only object can tell us about textures to use.
- Instead, just do like the json manager: fetch all the bitmap font data, and store them.

By doing this, you'll ensure that when `gdjs.BitmapFontManager.prototype.loadBitmapFont` is called, you can do two things:
- if the bitmap font is not **installed** (i.e: if the font is not in PIXI.BitmapFont.available), you call `PIXI.BitmapFont.install`. You have **already loaded** everything! The fontData is preloaded, and the texture is given to you (because it was preloaded by the ImageManager).
- if it's already installed/available, you just return it :) 

This also means that at runtime, we don't need async/await or promises there because we preloaded all that was needed.

In other words, make sure to see the distinction between:
- *loading*, in the sense *downloading* the **data** that you need => that's `preloadBitmapFontData`. 
- *loading* in the sense sending the data and the texture to `BitmapFont.install`. => that's `loadBitmapFont`. I actually recommend you rename this `getBitmapFont` or `getBitmapFontFromData` to clear any confusion.

Reply:
Note that I wrote also in another comment:

It might be a disappointment that we can't call `BitmapFont.install` during the preloading.

=> The problem is that the xml/fnt data file does NOT have the texture... so we can't! But maybe later we'll create resources that are "embedding other resources".

This will need some rework to make resources depend on other resources. Not trivial. For now, we're better with two different resources (texture, loaded by another manager, and bitmap font *data* loaded by this manager).